addons/mail/static/tests/owl/hooks/use_store_compare_keys_tests.js

```diff
@@ -0,0 +1,218 @@
+odoo.define('mail.hooks.useStoreCompareKeysTests', function (require) {
+'use strict';
+
+const { proxyComparator, proxyComparatorDeep, useStoreCompareKeys } = require('mail.hooks.useStoreCompareKeys');
+
+const { Component, QWeb, Store } = owl;
+const { xml } = owl.tags;
+
+const {
+    beforeEach: utilsBeforeEach,
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    nextAnimationFrame,
+} = require('mail.messagingTestUtils');
+
+QUnit.module('mail.messaging', {}, function () {
+QUnit.module('hooks', {}, function () {
+QUnit.module('useStoreCompareKeysTests', {
+    beforeEach() {
+        utilsBeforeEach(this);
+        const qweb = new QWeb();
+        this.env = { qweb };
+        this.start = () => {
+            this.store = new Store({
+                env: this.env,
+                state: {
+                    obj: {
+                        a: 1,
+                        b: 2,
+                        c: 3,
+                    },
+                },
+            });
+            this.env.store = this.store;
+            this.hashFn = this.store.observer.revNumber.bind(this.store.observer);
+            this.isEqual = (a, b) => a === b;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        this.env = undefined;
+        this.store = undefined;
+        this.hashFn = undefined;
+        this.isEqual = undefined;
+    }
+});
+
+QUnit.test("proxyComparator when strict equal", async function (assert) {
+    assert.expect(6);
+    this.start();
+    const comparator = proxyComparator(this.hashFn, this.isEqual);
```

Comment:
I would use a helper function, so that you don't have to always pass `hashFn` and `isEqual` (and consequently, no need them in `this`)

Reply:
Obsolete, not testing internal functions.